bridges/otelzap/encoder.go

```diff
@@ -158,14 +161,20 @@ type arrayEncoder struct {
 	elems []log.Value // nolint:unused
 }
 
-// TODO.
 func (a *arrayEncoder) AppendArray(v zapcore.ArrayMarshaler) error {
-	return nil
+	// TODO: Use arrayEncoder from a pool.
+	arr := &arrayEncoder{}
+	err := v.MarshalLogArray(arr)
+	a.elems = append(a.elems, log.SliceValue(arr.elems...))
+	return err
```

Comment:
Can you add a test that covers this scenario?

Reply:
It is covered by AddArray (nested) case

Reply:
This is not true. Run the tests with a debugger. Only `objectEncoder.AddArray`  is covered.